source/extensions/all_extensions.bzl

```diff
@@ -7,6 +7,8 @@ _required_extensions = {
     "envoy.common.crypto.utility_lib": "//source/extensions/common/crypto:utility_lib",
     "envoy.request_id.uuid": "//source/extensions/request_id/uuid:config",
     "envoy.transport_sockets.tls": "//source/extensions/transport_sockets/tls:config",
+    "envoy.listener.quic": "//source/extensions/quic_listeners/quiche:quic_factory_lib",
+    "envoy.transport_sockets.quic": "//source/extensions/quic_listeners/quiche:quic_transport_socket_factory_lib",
```

Comment:
Is your plan to follow up here and completely remove these as extensions? TODO? Also this will allow us to use the GSO writer for udp_proxy also and remove all of that config as well. TODO around that?

Reply:
hm, I didn't think having quic as a "required" extension made any less sense than tls as a required extension, but if you want to remove it entirely we can.  Want me to do that here, or a follow up?

I wasn't planning on following up on the GSO bits, I was largely hoping to just make the QUIC hackery I'm dealing with less hacky.  Would you or @danzh2010 be up for tackling GSO as a followup?

Reply:
> hm, I didn't think having quic as a "required" extension made any less sense than tls as a required extension, but if you want to remove it entirely we can. Want me to do that here, or a follow up?

TLS as an extension is "required" because of the openssl stuff. I think if we completely remove QUIC as an extension it will clean up a lot of the places I have commented on where we are referencing extension code in core code and vice versa. I'm happy to do some of this work as a side project if you don't have time but let's add the TODO either way?

> I wasn't planning on following up on the GSO bits, I was largely hoping to just make the QUIC hackery I'm dealing with less hacky. Would you or @danzh2010 be up for tackling GSO as a followup?

Yes I can work on this as part of ^ if you don't want to do it now.